core/site.go

```diff
@@ -484,13 +484,19 @@ func (site *Site) updateMeters() error {
 
 			// battery soc and capacity
 			var capacity float64
-			soc, err := soc.Guard(meter.(api.Battery).Soc())
+			capacity, capacityErr := meter.(api.BatteryCapacity).Capacity()
```

Comment:
This unchecked type assertion caused a crash on b7ac2ad201f2494fcc5ac4a139c237ac7a7a661c. I would propose we enable the `forcetypeassert` linter everywhere.

Reply:
Kannst Du das erklären? Warum sollte das crashen? Gerne separte Diskussion zu `forcetypeassert`.

Reply:
-> #9833

Reply:
Erklärt nicht, wo hier ein Crash entstehen kann?

Reply:
Wenn `meter` nicht die type assertion auf `api.BatteryCapacity` erfüllt gibt es hier eine panic. Dadurch, dass ich dieses statement außerhalb des `if socErr == nil {` gezogen hatte kam es zu der panic. Somit erfordert der Code implizit, dass alle meter die `api.BatteryCapacity` nicht erfüllen einen `socErr` produzieren.
Um diese ganze klasse an Fehler zu erkennen und zukünftig zu vermeiden habe ich den PR erstellt.